lighthouse-core/gather/gatherers/seo/tap-targets.js

```diff
@@ -0,0 +1,351 @@
+      // own size is 0x0 and there's no visible child content
+      return [];
+    }
+  }
+
+  // Treating overflowing content in scroll containers as invisible could mean that
+  // most of a given page is deemed invisible. But:
+  // - tap targets audit doesn't consider different containers/layers
+  // - having most content in an explicit scroll container is rare
+  // - treating them as hidden only generates false passes, which is better than false failures
+  clientRects = filterClientRectsWithinAncestorsVisibleScrollArea(node, clientRects);
+
+  return clientRects;
+}
+
+/**
+ *
+ * @param {Element} node
+ * @param {LH.Artifacts.Rect[]} clientRects
+ * @returns {LH.Artifacts.Rect[]}
+ */
+/* istanbul ignore next */
+function filterClientRectsWithinAncestorsVisibleScrollArea(node, clientRects) {
+  const parent = node.parentElement;
+  if (!parent) {
+    return clientRects;
+  }
+  if (getComputedStyle(parent).overflowY !== 'visible') {
+    const parentBCR = parent.getBoundingClientRect();
+    clientRects = clientRects.filter(cr => rectContains(parentBCR, cr));
+  }
+  if (parent.parentElement && parent.parentElement.tagName !== 'BODY') {
+    return filterClientRectsWithinAncestorsVisibleScrollArea(
+      parent,
+      clientRects
+    );
+  }
+  return clientRects;
+}
+
+/**
+ * @param {Element} node
+ * @param {boolean} includeChildren
+ * @returns {LH.Artifacts.Rect[]}
+ */
+/* istanbul ignore next */
+function getClientRects(node, includeChildren = true) {
+  /** @type {LH.Artifacts.Rect[]} */
+  let clientRects = Array.from(
+    node.getClientRects()
+  ).map(clientRect => {
+    // Contents of DOMRect get lost when returned from Runtime.evaluate call,
+    // so we convert them to plain objects.
+    const {width, height, left, top, right, bottom} = clientRect;
+    return {width, height, left, top, right, bottom};
+  });
+  if (includeChildren) {
+    for (const child of node.children) {
+      clientRects = clientRects.concat(getClientRects(child));
+    }
+  }
+
+  return clientRects;
+}
+
+/**
+ * @param {Element} node
+ * @returns {boolean}
+ */
+/* istanbul ignore next */
+function nodeHasParentTapTarget(node) {
+  if (!node.parentElement) {
+    return false;
+  }
+  if (node.parentElement.matches(tapTargetsSelector)) {
+    return true;
+  }
+  return nodeHasParentTapTarget(node.parentElement);
+}
+
+/**
+ * Check if node is in a block of text, such as paragraph with a bunch of links in it.
+ * Makes a reasonable guess, but for example gets it wrong if the element is surounded by other
+ * HTML elements instead of direct text nodes.
+ * @param {Node} node
+ * @returns {boolean}
+ */
+/* istanbul ignore next */
+function nodeIsInTextBlock(node) {
+  /**
+   * @param {Node} node
+   * @returns {boolean}
+   */
+  function isInline(node) {
+    if (node.nodeType === Node.TEXT_NODE) {
+      return true;
+    }
+    if (node.nodeType !== Node.ELEMENT_NODE) {
```

Comment:
if an `Element`, these two cases can't ever happen. If we can delete those, might be cleaner to just drop the `isInline` function and inline the `getComputedStyle` check into the if statement where it's used below